doc/rados/troubleshooting/troubleshooting-mon.rst

```diff
@@ -428,6 +428,36 @@ or::
 
   Corruption: 1 missing files; e.g.: /var/lib/ceph/mon/mon.foo/store.db/1234567.ldb
 
+
+Symptoms of a constantly growing monitor
+----------------------------------------
+
+If the cluster is unhealty, every osdmap is stored until the cluster reaches a healthy state.
+In all other cases, the monstore should not grow beyond sane levels of a few gigabytes, even on large clusters.
+In cases of internal counter corruptions, old versions of objects don't get garbage-collected, causing
```

Comment:
Do you have more information about what the nature of this corruption was and what led to it? I understand that this tool can be helpful to get us out of the situation that you ran into, but it also seems like an invariant was broken somewhere.

Reply:
I have no idea. I guess we had a power outage or a daemon/kernel crash which caused the first and last counters to be nonsensical, maybe rocksdb transaction problems.
Parts of our corruption could even be very old in some older tables that are hardly written into.
This might be because of bad implementation of write barrier on the underlying hardware, but I have nothing to pinpoint. I just observed the phenoma after it grew out of proportion and investigated. After some attempts I had a repair routine that worked to fit the cluster.
In theory, some of this logic could be implemented in the mon server, but I went the conservative approach to repair a corruption. I'm not that much in to the interna of ceph for fixing it live.

Reply:
interesting.  Back on Firefly I had mon DB inflation issues but this is new.  I also echo Neha's thoughts above.